Formula/retroshare-service.rb

```diff
@@ -0,0 +1,52 @@
+class RetroshareService < Formula
+  desc "RetroShare friend to friend communication headless service"
+  homepage "https://retroshare.cc/"
+  url "https://github.com/RetroShare/RetroShare/releases/download/v0.6.6/RetroShare-v0.6.6-497-g2b4e57133-homebrew.tar.gz"
```

Comment:
What's the difference between the homebrew tar and the default one?

Reply:
It is based on much newer code-base, that 497 is to be read 497 commits after. This includes lot of bug fixes the CMake based build system and a bunch of fixes to make it compile on MacOS like this https://github.com/RetroShare/libretroshare/commit/a28ec6d8a81581e330dcffc01197d2449e9dd3c0

Reply:
To have a "special" release for homebrew is consensual with upstream development team, and will not be necessary anymore for retroshare 0.6.7 but there is not yet plan for that release so a lot of time will pass before it.

Reply:
It is a chicken and egg problem, we already had a similar discussion from the opposite perspective for RetroShare-gui 0.6.5 release https://github.com/Homebrew/homebrew-core/pull/40970 it needed a few changes to compile well on homebrew environment and that wasn't acceptable at that time. We are a little team with few resources to support so many different platforms especially some with so many economical restrictions, we have many MacOS users, just a few of them are capable of compiling from source, so the others ends up using version compiled/crafted by mostly unknown parties. We have a long history and I would not consider us a "niche" project, we managed to had MacOS releases in the past but that have become increasingly difficult over the years due to Apple very restrictive licensing, and non retro-compatible changes even in common C/C++ headers. We get lot of reports from them in our internal forum, so even this changes from a few days got tested by them but we lack a way to properly distribute the software in a trusted and sustainable manner. We really like homebrew project and it seems the best fit for our situation, after discovering that it works on Linux too I am capable of fiddling with the formula on my laptop too, but that is only for the Linux build, so I am reliant on this PR and community reports for MacOS.